chai_lab/data/parsing/input_validation.py

```diff
@@ -0,0 +1,73 @@
+"""
+Simple heuristics that can help with identification of EntityType
+"""
+
+import string
+from string import ascii_letters
+
+from chai_lab.data.parsing.structure.entity_type import EntityType
+
+
+def constituents_of_modified_fasta(x: str) -> list[str] | None:
+    """
+    Accepts RNA/DNA inputs: 'agtc', 'AGT[ASP]TG', etc
+    Returns constituents, e.g, [A, G, T, ASP, T, G] or None if string is incorrect
+    """
+    x = x.strip().upper()
+    # it is a bit strange that digits are here, but [NH2] was in one protein
+    allowed_chars = ascii_letters + "[]" + string.digits
+    if not all(letter in allowed_chars for letter in x):
+        return None
+
+    current_modified: str | None = None
+
+    constituents = []
+    for letter in x:
+        if letter == "[":
+            if current_modified is not None:
+                return None  # double open bracket
+            current_modified = ""
+        elif letter == "]":
```

Comment:
Let's change this and the opening brace char to "(" to be more in line with how mmCIF contains modified residues.